packages/moxb/src/routing/url-arg/__tests__/UrlArgTypeUnorderedStringArray.test.ts

```diff
@@ -18,8 +18,28 @@ describe('URL arg type boolean', () => {
         expect(v1).toBe(v2); // The objects are actually the same, not only equal!
     });
 
+    it('should return the same array instance on subsequent calls with the same string for the same param', () => {
+        const v1 = typeDef.getParser('test')('a,b,c', ['42']);
+        const v2 = typeDef.getParser('test')('a,b,c', ['42']);
+        expect(v1).toBe(v2); // The objects are actually the same, not only equal!
+    });
+
+    it('should return the same array instance independent of order', () => {
+        // To be consistent, this should also be the same not equal! (else we can drop the entire sameness)...
+        const v1 = typeDef.getParser('test')('a,b,c', ['42']);
+        const v2 = typeDef.getParser('test')('c,a,b', ['42']);
+        expect(v1).toBe(v2); // The objects are actually the same, not only equal!
+    });
+
+    it('should return the same array instance on subsequent calls with the same string for the different params', () => {
+        const v1 = typeDef.getParser('test')('a,b,c', ['42']);
+        const v2 = typeDef.getParser('x')('a,b,c', ['42']);
+        expect(v1).not.toBe(v2); // The objects are actually not the same
+        expect(v1).toEqual(v2); // ...but equal!
+    });
+
     it('should be able to parse empty strings', () => {
-        expect(parser('', ['42'])).toEqual([]);
+        expect(parser('', ['42'])).toEqual(['42']);
     });
```

Comment:
As I described above, I don't think the parsers need to be able to handle empty strings.